web/src/components/tasks/government-contracting/GovernmentContractingPaginator.tsx

```diff
@@ -0,0 +1,134 @@
+  const [stepIndex, setStepIndex] = useState(0);
+  const [quickActionTask, setQuickActionTask] = useState<QuickActionTask>();
+  const { Config } = useConfig();
+
+  useEffect(() => {
+    const getQuickActionTaskObj = async (): Promise<QuickActionTask> => {
+      return await fetchQuickActionByFilename(stepNumberToFilename[stepIndex]);
+    };
+
+    getQuickActionTaskObj().then((quickActionTask) => {
+      setQuickActionTask(quickActionTask);
+    });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [stepIndex]);
+
+  const stepNumberToFilename: Record<number, string> = {
+    0: "njstart-registration",
+    1: "apprenticeship-program-intake",
+    2: "public-works-contractor-registration",
+    3: "prevailing-wage",
+  };
+
+  const onMoveToStep = (stepIndex: number): void => {
+    setStepIndex(stepIndex);
+  };
+
+  const getNextButtonText = (stepIndex: number): string => {
+    if (stepIndex === GovernmentContractingStepsConfiguration.length - 1) {
+      return Config.taskDefaults.backButtonText;
+    } else {
+      return Config.taskDefaults.continueButtonText;
+    }
+  };
+
+  const displayButtons = (): ReactNode => {
+    return (
+      <div className="margin-top-2 ">
+        <div className="bg-base-lightest padding-3 margin-top-3 margin-bottom-neg-4 radius-bottom-lg">
+          <ReverseOrderInMobile className={"flex-justify-end"}>
+            <>
+              {shouldDisplayPreviousButton(stepIndex) && (
+                <div className="margin-top-1 mobile-lg:margin-top-0 mobile-lg:margin-right-105">
+                  <SecondaryButton
+                    isColor="primary"
+                    onClick={(): void => {
+                      onMoveToStep(stepIndex - 1);
+                    }}
+                    dataTestId="previous-button"
+                    isRightMarginRemoved={true}
+                  >
+                    {Config.taskDefaults.backButtonText}
+                  </SecondaryButton>
+                </div>
+              )}
+              {shouldDisplayContinueButton(stepIndex) && (
+                <PrimaryButton
+                  isColor="primary"
+                  onClick={(): void => {
+                    onMoveToStep(stepIndex + 1);
+                  }}
+                  isRightMarginRemoved={true}
+                  dataTestId="next-button"
+                >
+                  {getNextButtonText(stepIndex)}
+                </PrimaryButton>
+              )}
+            </>
+          </ReverseOrderInMobile>
+        </div>
+      </div>
+    );
+  };
+
+  const shouldDisplayPreviousButton = (stepIndex: number): boolean => {
+    return stepIndex !== 0;
+  };
+
+  const shouldDisplayContinueButton = (stepIndex: number): boolean => {
+    return stepIndex !== GovernmentContractingStepsConfiguration.length - 1;
+  };
+
+  const stepStates = GovernmentContractingStepsConfiguration.map((value) => {
+    return {
+      name: value.name,
+    };
+  });
```

Comment:
[sand] This can also be moved outside the component. And as for `shouldDisplayPreviousButton` and `shouldDisplayContinueButton`, perhaps these could be self-contained components so that we can reduce the amount of logic in the display of the component?